### PR TITLE
PixFuture Bid Adapter: add new bid adapter

### PR DESCRIPTION
**Pull request by pixfuture-media**
## Type of change
- [ ] New PixFuture Media Inc bidder adapter  

## Description of change

New PixFuture Media Inc bidder adapter

- test parameters for validating bids
```
var adUnits = [{
   {
  "pageUrl": "some referer URL",
  "bidId": "608601dbbacc5",
  "auctionId": "4842f47b-a12b-4eed-8eac-fc71d37a5e30",
  "transactionId": "b4004949-b099-48eb-ba04-902f4fffcaca",
  "adUnitCode": "26335x300x250x14x_ADSLOT88",
  "sizes": [
    [300, 250],
    [300, 600]
  ],
  {
  "params": {
    "pix_id": "777"
  }];
```

- contact email of the adapter’s maintainer
- [admin@pixfuture.net ] official adapter submission

### CODE REVIEW

**Comment by MartianTribe** [comment]
Docs PR:
https://github.com/prebid/prebid.github.io/pull/3034

File: modules/pixfutureBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+    return validBidRequests.map((bidRequest) => {
+      let referer = '';
+      if (bidderRequest && bidderRequest.refererInfo) {
+        referer = bidderRequest.refererInfo.referer || '';
+      }
+
+      const userObjBid = find(validBidRequests, hasUserInfo);
+      let userObj = {};
+      if (config.getConfig('coppa') === true) {
+        userObj = {'coppa': true};
```
Comment (FilipStamenkovic) [inline modules/pixfutureBidAdapter.js:43]: You didn't add `coppa` support in your docs PR.

File: modules/pixfutureBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+        sdk: {
+          source: SOURCE,
+          version: '$prebid.version$'
+        },
+        schain: schain
+      };
+
+      if (bidderRequest && bidderRequest.uspConsent) {
+        payload.us_privacy = bidderRequest.uspConsent
+      }
```
Comment (FilipStamenkovic) [inline modules/pixfutureBidAdapter.js:81]: You didn't add `usp` support in your docs PR.

File: modules/pixfutureBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+        let eids = [];
+
+        addUserId(eids, utils.deepAccess(validBidRequests[0], `userId.flocId.id`), 'chrome.com', null);
+        addUserId(eids, utils.deepAccess(validBidRequests[0], `userId.criteoId`), 'criteo.com', null);
+        addUserId(eids, utils.deepAccess(validBidRequests[0], `userId.unifiedId`), 'thetradedesk.com', null);
+        addUserId(eids, utils.deepAccess(validBidRequests[0], `userId.id5Id`), 'id5.io', null);
+        addUserId(eids, utils.deepAccess(validBidRequests[0], `userId.sharedId`), 'thetradedesk.com', null);
+        addUserId(eids, utils.deepAccess(validBidRequests[0], `userId.identityLink`), 'liveramp.com', null);
+        addUserId(eids, utils.deepAccess(validBidRequests[0], `userId.liveIntentId`), 'liveintent.com', null);
+        addUserId(eids, utils.deepAccess(validBidRequests[0], `userId.fabrickId`), 'home.neustar', null);
```
Comment (FilipStamenkovic) [inline modules/pixfutureBidAdapter.js:103]: In your docs PR you added `userIds: all`, but here you don't pull **all** userIds, just couple of them. Either correct your docs PR, or add support for all of them.

File: modules/pixfutureBidAdapter.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  "transactionId": "b4004949-b099-48eb-ba04-902f4fffcaca",
+  "adUnitCode": "26335x300x250x14x_ADSLOT88",
+  "sizes": [
+    [300, 250],
+    [300, 600]
+  ],
+  {
+  "params": {
+    "pix_id": "777"
+  }];
```
Comment (FilipStamenkovic) [inline modules/pixfutureBidAdapter.md:28]: This config example doesn't work, please correct it. You can easily verify your config example in `integrationExamples` (you could use `/gpt/hello_world.html`).

**Review by FilipStamenkovic — CHANGES_REQUESTED** [review]
Beside thing already mentioned in other comments, your test coverage is low, it should be at least 80%.  
Run:

```
gulp test-coverage
gulp view-coverage
```

Also, since pbjs 5.0, it is required for your bid adapter to set: `meta.advertiserDomains`. You can take a look at some other bidder to see how they are handling it.

File: modules/pixfutureBidAdapter.js (hunk 1 above)
Comment (pixfuture-media) [reply modules/pixfutureBidAdapter.js:43]: The `coppa` has been added to docs.
Also added added `meta.advertiserDomains` to the pixfutureBidAdapter.js

File: modules/pixfutureBidAdapter.js (hunk 2 above)
Comment (pixfuture-media) [reply modules/pixfutureBidAdapter.js:81]: The  `usp` support has been added to docs PR

File: modules/pixfutureBidAdapter.js (hunk 3 above)
Comment (pixfuture-media) [reply modules/pixfutureBidAdapter.js:103]: In the docs PR has been fixed list of `userIds`

File: modules/pixfutureBidAdapter.md (hunk 4 above)
Comment (pixfuture-media) [reply modules/pixfutureBidAdapter.md]: The config example has been fixed

**Comment by pixfuture-media** [comment]
> Beside thing already mentioned in other comments, your test coverage is low, it should be at least 80%.
> Run:
> 
> ```
> gulp test-coverage
> gulp view-coverage
> ```
> 
> Also, since pbjs 5.0, it is required for your bid adapter to set: `meta.advertiserDomains`. You can take a look at some other bidder to see how they are handling it.

**Comment by pixfuture-media** [comment]
> > Beside thing already mentioned in other comments, your test coverage is low, it should be at least 80%.
> > Run:
> > ```
> > gulp test-coverage
> > gulp view-coverage
> > ```

This two test gulp test-coverage and gulp view-coverage give to us empty info. Can you help with this issue?
Please see image in the attachment
Screenshot 2021-06-21 171754

**Comment by FilipStamenkovic** [comment]
@pixfuture-media you closed this PR on purpose?

**Pull request closed without merging**